marconi-chain-index/src/Marconi/ChainIndex/Experimental/Runner.hs

```diff
@@ -43,48 +50,41 @@ given indexer.
 If you want to start several indexers, use @runIndexers@.
 -}
 runIndexer
-  :: ( Core.IsIndex (ExceptT Core.IndexerError IO) (WithDistance BlockEvent) indexer
+  :: ( WithDistance BlockEvent ~ event
+     , Core.IsIndex (ExceptT Core.IndexerError IO) event indexer
      , Core.Closeable IO indexer
      )
   => Trace IO Text
+  -> SecurityParam
   -> RetryConfig
   -> FilePath
   -> C.NetworkId
-  -> C.ChainPoint
+  -> [C.ChainPoint]
   -> indexer (WithDistance BlockEvent)
   -> IO ()
-runIndexer trace retryConfig socketPath networkId _startingPoint indexer = do
+runIndexer trace securityParam retryConfig socketPath networkId startingPoints indexer = do
   withNodeConnectRetry trace retryConfig socketPath $ do
-    securityParam <- Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
+    Trace.logInfo trace $
+      PP.renderStrict $
+        PP.layoutPretty PP.defaultLayoutOptions $
+          PP.pretty $
+            LastSyncPointsLog startingPoints
```

Comment:
extract it in a function to improve readability

Reply:
I think this should be done in a separate PR. Ideally, we would have a typeclass (`class HasLog`) which does this. Something like what RIO this https://hackage.haskell.org/package/rio-0.0.3.0/docs/RIO-Prelude-Logger.html does.

What do you think?

Reply:
I think this should be done in a separate PR. Ideally, we would have a typeclass (`class HasLog`) which does this. Something like what RIO this https://hackage.haskell.org/package/rio-0.0.3.0/docs/RIO-Prelude-Logger.html does.

What do you think?

Reply:
We may talk about two different things here. I commented on the noise introduced to enable the `PrettyPrinter` resolution.

Reply:
We always _had_ to do this when using `logInfo` with `PrettyPrinter`. Check below with `logError`. Or in `Marconi.ChainIndex.Node.Client.Retry`. I agree it adds noise, but I think this should be done in a separate PR in order to do the change everywhere in the codebase.